Makefile

```diff
@@ -584,14 +584,15 @@ rpkg:
 	cp -rf 3rdparty/tvm/nnvm/include/* R-package/inst/include
 	Rscript -e "if(!require(devtools)){install.packages('devtools', repo = 'https://cloud.r-project.org/')}"
 	Rscript -e "library(devtools); library(methods); options(repos=c(CRAN='https://cloud.r-project.org/')); install_deps(pkg='R-package', dependencies = TRUE)"
-	echo "import(Rcpp)" > R-package/NAMESPACE
-	echo "import(methods)" >> R-package/NAMESPACE
+	cp R-package/dummy.NAMESPACE R-package/NAMESPACE
```

Comment:
Can you point me to some source on how `R CMD INSTALL` deal with `dummy.NAMESPACE`? Seems it is automatically ignored.

Reply:
the NAMESPACE defines the exported symbols of the package. It is read on loading of package by R runtime.
To clarify the move to "dummy.NAMESPACE": the original code generated a very simple NAMESPACE (as you see in the removed lines) by pasting those 2 lines, which my have sufficed with legacy versions.
Modern build tools also check the complete package+install cycle and then detects missing symbols, so the intermediate build fails, this build is necessary to setup the infrastructure to generate the final NAMESPACE file.
This is the reason, why the CMD INSTALL is run twice, so my understanding.
Only the NAMESPACE of the second CMD INSTALL is used by final package.